### PR TITLE
Remove deprecated ray.worker.get_resource_ids()

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
This api is already marked as deprecated.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — COMMENTED** [review]
Could we `raise DeprecationWarning()` instead of outright removing? This will provide a smoother transition path.

In general I think we want to go from mark / log deprecated -> raise DeprecationWarning -> completely removed.

File: python/ray/worker.py [hunk 1]
```diff
@@ -566,15 +565,7 @@ def get_resource_ids():
...
-    worker = global_worker
-    worker.check_connected()
-
-    if _mode() == LOCAL_MODE:
-        raise RuntimeError(
-            "ray.worker.get_resource_ids() currently does not work in local_mode."
-        )
-
-    return global_worker.core_worker.resource_ids()
+    raise DeprecationWarning("get_resource_ids() is deprecated")
```
Comment (jjyao) [inline python/ray/worker.py:568]: Is there any public alternative we should suggest here? @ericl

Comment (ericl) [reply python/ray/worker.py:568]: get_gpu_ids() I think (cc @wuisawesome )

Comment (wuisawesome) [reply python/ray/worker.py:568]: Yup, GPUs are the only resource with ids. (Changing this config isn't a public api) https://sourcegraph.com/github.com/ray-project/ray/-/blob/src/ray/common/ray_config_def.h?L550

**Review by ericl — APPROVED** [review]

**Comment by ericl** [comment]
We've deprecated the entire module; so this PR is no longer needed specifically to remove this method.

**Pull request closed without merging**